### ISSUE TITLE
schema evolution and Schema Registry

### ISSUE THREAD

**Issue by Arnold1** [actgardner/gogen-avro#136]
Hi,

how does gogen-avro support schema evolution and Schema Registry?

Thanks

**Comment by actgardner** [comment]
If you're referring the Confluent Schema Registry, there's an open issue to add support: https://github.com/actgardner/gogen-avro/issues/86

If you have both the reader and writer schemas, you can use the generated `Deserialize<Type>` method for the reader schema, and pass in the writer schema as the second argument: https://github.com/actgardner/gogen-avro/blob/master/v7/test/evolve-union/schema_test.go#L22
The result will be the reader schema struct, populated with data written by the writer schema.

If you want to improve performance for deserializing a lot of records with the same schema, you can compile the VM bytecode once and reuse it like this (this is a template so the types aren't interpolated): https://github.com/actgardner/gogen-avro/blob/master/v7/generator/flat/templates/record_container.go#L23-L45

Basically:
```
        // Do this once for a given (reader, writer) pair
	deser, err := compiler.CompileSchemaBytes(<writer schema>, <reader schema>)
        
         // Do this for every record you're deserializing
         err := vm.Eval(<io.Reader>, deser, <a pointer to the target struct>)
```

**Comment by casimcdaniels** [comment]
@actgardner Happy Friday! 

Just curious what the suggestion is for usage of this library with consideration of schema registry (confluent  for example). Is this library necessary or serve as a replacement for schema registry? Trying to wrap my head around the ecosystem and I'm wondering about the intended workflow with this library versus using SR.
